sycl/test/check_device_code/device_asan_no_ctor.cpp

```diff
@@ -11,13 +11,6 @@
 
 using namespace sycl;
 
-int main() {
-  sycl::queue queue;
-  {
-    queue.submit(
-        [&](sycl::handler &cgh) { cgh.single_task<class t>([=]() {}); });
-    queue.wait();
-  }
-
-  return 0;
-}
+SYCL_EXTERNAL void nothing() {
+  // Intentionally empty
+}
```

Comment:
I think we need to drop changes in this test, at least for this round of changes.

Reply:
I've reverted the changes, although for my understanding's sake, can I ask why?

Reply:
I'm not sure if the transformation works on device functions or device kernels. If the latter, then the modified test didn't really go through the intended code path.